src/tiled/commandmanager.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * commandmanager.cpp
+ * Copyright 2017, Ketan Gupta <ketan19972010@gmail.com>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "commandmanager.h"
+
+#include "commanddatamodel.h"
+#include "commanddialog.h"
+
+#include <QApplication>
+#include <QAction>
+#include <QLatin1String>
+#include <QMenu>
+
+namespace Tiled {
+namespace Internal {
+
+CommandManager *CommandManager::mInstance;
+
+CommandManager::CommandManager()
+    : mModel(new CommandDataModel)
+{
+    
+}
+
+CommandManager *CommandManager::instance()
+{
+    if (!mInstance)
+        mInstance = new CommandManager;
+
+    return mInstance;
+}
+
+void CommandManager::deleteInstance()
+{
+    delete mInstance;
+    mInstance = nullptr;
+}
+
+CommandDataModel *CommandManager::getCommandDataModel()
```

Comment:
Naming style: getters do not have a "get" prefix, this should just be `commandDataModel`.